flash/image/classification/input_transform.py

```diff
@@ -51,7 +51,13 @@ def input_per_sample_transform(self):
 
     def train_input_per_sample_transform(self):
         return T.Compose(
-            [T.ToTensor(), T.Resize(self.image_size), T.Normalize(self.mean, self.std), T.RandomHorizontalFlip()]
+            [
+                T.ToTensor(),
+                T.Resize(self.image_size),
+                T.Normalize(self.mean, self.std),
+                T.RandomHorizontalFlip(),
```

Comment:
Do we really need the `RandomHorizontalFlip` in this case when `TrivialAugmentWide` is applied before hand?

Reply:
not sure what geometrical transformation is applied, check some samples and most changes were about color intensities...

Reply:
That's a good question, @ligaz - I was looking at the source code of this transform. Here are the transforms it applies, if I'm not wrong:

https://github.com/pytorch/vision/blob/62740807c18e68bb0acd85895dca527f9a655bd5/torchvision/prototype/transforms/_auto_augment.py#L428:L449

```python
    _AUGMENTATION_SPACE = {
        "Identity": (lambda num_bins, height, width: None, False),
        "ShearX": (lambda num_bins, height, width: torch.linspace(0.0, 0.99, num_bins), True),
        "ShearY": (lambda num_bins, height, width: torch.linspace(0.0, 0.99, num_bins), True),
        "TranslateX": (lambda num_bins, height, width: torch.linspace(0.0, 32.0, num_bins), True),
        "TranslateY": (lambda num_bins, height, width: torch.linspace(0.0, 32.0, num_bins), True),
        "Rotate": (lambda num_bins, height, width: torch.linspace(0.0, 135.0, num_bins), True),
        "Brightness": (lambda num_bins, height, width: torch.linspace(0.0, 0.99, num_bins), True),
        "Color": (lambda num_bins, height, width: torch.linspace(0.0, 0.99, num_bins), True),
        "Contrast": (lambda num_bins, height, width: torch.linspace(0.0, 0.99, num_bins), True),
        "Sharpness": (lambda num_bins, height, width: torch.linspace(0.0, 0.99, num_bins), True),
        "Posterize": (
            lambda num_bins, height, width: cast(torch.Tensor, 8 - (torch.arange(num_bins) / ((num_bins - 1) / 6)))
            .round()
            .int(),
            False,
        ),
        "Solarize": (lambda num_bins, height, width: torch.linspace(255.0, 0.0, num_bins), False),
        "AutoContrast": (lambda num_bins, height, width: None, False),
        "Equalize": (lambda num_bins, height, width: None, False),
    }
```

IMO, we could have dropped horizontal flip since this transform does have enough augmentations, _but_, since it's not available for all torchvision versions, so I'll honestly put a condition.

If this augmentation is available, use it, else include `RandomHorizontalFlip`.

```python
transforms = [
    T.ToTensor(),
    T.Resize(self.image_size),
    T.Normalize(self.mean, self.std)
]

transforms += [T.TrivialAugmentWide()] if _TORCHVISION_GREATER_EQUAL_0_11 else [T.RandomHorizontalFlip()]
```

**Note**: I think it's a better idea to convert the input to a tensor first, resize and normalize before applying this transform?Let me know what you guys think. @Borda @ligaz